### PR TITLE
Implement 64 bits 1 sample delta compression case

### PR DESCRIPTION
**Pull request by ameykasbe**
### Summary
* The handling of the one sample of the i64 delta case has been implemented.
* This change enhances the robustness of our compression algorithm and ensures that it can handle a wider range of cases.
* Updated and added unit tests.

### CODE REVIEW

File: tsz-compress/src/v2/encode.rs [hunk 1]
```diff
@@ -159,6 +163,9 @@ impl EmitDeltaBits for CompressionQueue<10> {
         // Check if the values will fit in the cases
         let values = self.peak_bitcounts::<10>();
         for (index, bits_required) in values.into_iter().enumerate() {
+            if (index < 2) & (bits_required > 32) {
```
Comment (trueb2) [inline tsz-compress/src/v2/encode.rs]: This adds a very uncommon branch to the hot loop. We could move this check into the handling for the 32 bit case, which is already last/least likely. Branches were one of the main performance hits on Cortex-M4

**Review by trueb2 — CHANGES_REQUESTED** [review]
This PR adds i64 delta support and updates the default for i32 deltas to be computed on i64 bit width. This increases correctness but comes at some cost for compression. 

To best serve embedded compressors, we will proceed under the assumption that deltas are rarely outside of i32::MIN..=i32::MAX. A compressor on a 32-bit system should use the field attribute `#[tsz(delta = "i32")]` to use i32 bit width deltas at no additional cost.

File: tsz-compress/src/v2/encode.rs (hunk 1 above)
Comment (ameykasbe) [reply tsz-compress/src/v2/encode.rs]: The uncommon branch in the hot loop is removed in the commit: [7c2fe37](https://github.com/qsib-cbie/tsz/pull/46/commits/7c2fe37da02bf8cff7454fe80787e4460a0d3e61).

**Review by trueb2 — APPROVED** [review]
LGTM

**Pull request merged**